config/300-cloudbuildsource.yaml

```diff
@@ -0,0 +1,132 @@
+      ]
+  name: cloudbuildsources.events.cloud.google.com
+spec:
+  group: events.cloud.google.com
+  version: v1alpha1
+  names:
+    categories:
+    - all
+    - knative
+    - cloudbuildsource
+    - sources
+    kind: CloudBuildSource
+    plural: cloudbuildsources
+  scope: Namespaced
+  subresources:
+    status: {}
+  additionalPrinterColumns:
+    - name: Ready
+      type: string
+      JSONPath: ".status.conditions[?(@.type==\"Ready\")].status"
+    - name: Reason
+      type: string
+      JSONPath: ".status.conditions[?(@.type==\"Ready\")].reason"
+    - name: Age
+      type: date
+      JSONPath: .metadata.creationTimestamp
+  validation:
+    openAPIV3Schema:
+      properties:
+        spec:
+          required:
+            - sink
+          properties:
+            secret:
+              type: object
+              description: "Credential used to poll the Cloud Pub/Sub Subscription. It is not used to create or delete the Subscription, only to poll it. The value of the secret entry must be a service account key in the JSON format (see https://cloud.google.com/iam/docs/creating-managing-service-account-keys). Defaults to secret.name of 'google-cloud-key' and secret.key of 'key.json'."
+            project:
+              type: string
+              description: "ID of the Google Cloud Project that the Pub/Sub Topic exists in. E.g. 'my-project-1234' rather than its display name, 'My Project' or its number '1234567890'. If omitted uses the Project ID from the GKE cluster metadata service."
+            sink:
+              type: object
+              description: "Reference to an object that will resolve to a domain name to use as the sink."
+              anyOf:
+                - type: object
+                  properties:
+                    uri:
+                      type: string
+                      minLength: 1
+                - type: object
+                  properties:
+                    ref:
+                      type: object
+                      required:
+                        - apiVersion
+                        - kind
+                        - name
+                      properties:
+                        apiVersion:
+                          type: string
+                          minLength: 1
+                        kind:
+                          type: string
+                          minLength: 1
+                        name:
+                          type: string
+                          minLength: 1
+            ceOverrides:
+              type: object
+              description: "Defines overrides to control modifications of the event sent to the sink."
+              properties:
+                extensions:
+                  type: object
+                  description: "Extensions specify what attribute are added or overridden on the outbound event. Each `Extensions` key-value pair are set on the event as an attribute extension independently."
+            topic:
```

Comment:
do we need `topic` here if it always will be `cloud_builds`? Maybe this is something we can just hardcode and don't burden the user to configure...

Reply:
It's always "cloud-builds". The topic field is optional. If user doesn't set it, we will set it to "cloud-builds". If user sets it, it has to be "cloud-builds", otherwise it won't pass the validation.